radix-engine/src/system/system.rs

```diff
@@ -2877,8 +2881,17 @@ where
         public_keys: &[Bls12381G1PublicKey],
         signature: &Bls12381G2Signature,
     ) -> Result<u32, RuntimeError> {
-        // TODO costing
-        Ok(aggregate_verify_bls12381_v1(messages, public_keys, signature) as u32)
+        if !messages.is_empty() && !public_keys.is_empty() {
+            self.api.kernel_get_system().modules.apply_execution_cost(
+                ExecutionCostingEntry::Bls12381V1AggregateVerify {
+                    size: messages.iter().map(|m| m.len()).sum::<usize>() / messages.len(),
```

Comment:
Instead of using the average length of messages, would it be better to cost for each message and public key pair individually? 

This comes from a fear that somebody might have something like 9999x 1 byte long messages and 1x 1,000,000 byte long message which averages down to 100 bytes.

Reply:
Of course it would be better. But not sure if it adds up this way in case of aggregated verify. Let me check.

Reply:
Took a deeper look into this and discovered that used approach is not accurate enough in many cases, which could result in costing for less instructions count.
Additionally, if the messages sizes differ significantly the dispersion is even greater.
So, assuming that aggregated verify is a nice-to-have feature, I've decided to do not take aggregation into account and just sum up the costs per all message sizes. This returns higher instructions count. This way we will never charge for less instructions count.

This is a temporary approach until no better solution is found.